driver/reduceextreme_driver.hpp

```diff
@@ -0,0 +1,510 @@
+    indice     = std::vector<int32_t>(out_sz, static_cast<int32_t>(0));
+    indicehost = std::vector<int32_t>(out_sz, static_cast<int32_t>(0));
+
+    for(int32_t i = 0; i < in_sz; ++i)
+    {
+        x[i] = prng::gen_A_to_B<Tgpu>(static_cast<Tgpu>(-1.0), static_cast<Tgpu>(1.0));
+    }
+
+    if(x_dev->ToGPU(GetStream(), x.data()) != 0)
+        std::cerr << "Error copying (x) to GPU, size: " << x_dev->GetSize() << std::endl;
+
+    if(indice_dev->ToGPU(GetStream(), indice.data()) != 0)
+        std::cerr << "Error copying (indice) to GPU, size: " << indice_dev->GetSize() << std::endl;
+
+    if((reduceExtremeOp == MIOPEN_REDUCE_EXTREME_MIN) ||
+       (reduceExtremeOp == MIOPEN_REDUCE_EXTREME_MAX))
+    {
+        y_dev = std::unique_ptr<GPUMem>(new GPUMem(ctx, out_sz, sizeof(Tgpu)));
+        y     = std::vector<Tgpu>(out_sz, static_cast<Tgpu>(0));
+        yhost = std::vector<Tref>(out_sz, static_cast<Tref>(0));
+
+        if(y_dev->ToGPU(GetStream(), y.data()) != 0)
+            std::cerr << "Error copying (y) to GPU, size: " << y_dev->GetSize() << std::endl;
+    }
+
+    return miopenStatusSuccess;
+}
+
+template <typename Tgpu, typename Tref>
+int ReduceExtremeDriver<Tgpu, Tref>::RunForwardGPU()
+{
+    float kernel_total_time = 0;
+    float kernel_first_time = 0;
+
+    Timer t;
+    START_TIME
+
+    for(int32_t i = 0; i < inflags.GetValueInt("iter"); ++i)
+    {
+        if((reduceExtremeOp == MIOPEN_REDUCE_EXTREME_MIN) ||
+           (reduceExtremeOp == MIOPEN_REDUCE_EXTREME_MAX))
+        {
+            miopenReduceExtremeForward(GetHandle(),
+                                       xDesc,
+                                       x_dev->GetMem(),
+                                       dim,
+                                       reduceExtremeOp,
+                                       yDesc,
+                                       y_dev->GetMem(),
+                                       indiceDesc,
+                                       indice_dev->GetMem());
+        }
+        else
+        {
+            miopenReduceExtremeForward(GetHandle(),
+                                       xDesc,
+                                       x_dev->GetMem(),
+                                       dim,
+                                       reduceExtremeOp,
+                                       nullptr,
+                                       nullptr,
+                                       indiceDesc,
+                                       indice_dev->GetMem());
+        }
+
+        float time = 0;
+        miopenGetKernelTime(GetHandle(), &time);
+        kernel_total_time += time;
+        if(i == 0)
+            kernel_first_time = time;
+    }
+
+    if(inflags.GetValueInt("time") == 1)
+    {
+        STOP_TIME
+        int32_t iter = inflags.GetValueInt("iter");
+        if(WALL_CLOCK)
+            std::cout << "Wall-clock Time Forward ReduceExtreme Elapsed: " << t.gettime_ms() / iter
+                      << " ms\n";
```

Comment:
Please find all uses of `\n` in stream-based IO and replace with `std:endl`.

Reply:
fixed it.

Reply:
[resolved]